### PR TITLE
VideoPress: Add VideoStorageMeter component

### LINKED ISSUE #25751: VideoPress components: VideoStorageMeter
Image

### PR DESCRIPTION
**Pull request by dhasilva**
Closes #25751

#### Changes proposed in this Pull Request:

* Add `VideoStorageMeter` and `ProgressBar` components to VideoPress Package
* Add js tests to VideoPress Package
* Fix https://github.com/Automattic/jetpack/pull/25876#pullrequestreview-1087535452

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

N/A

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Run `cd projects/js-packages/storybook && pnpm run storybook:dev`
* Confirm that the default story of `Packages/VideoPress/Video Storage Meter` component works as expected
* Confirm that the default story of `Packages/VideoPress/Progress Bar` component works as expected
* Run `jetpack test packages/videopress js` and confirm test passes

Screenshot_2022-08-30_14-41-25

### CODE REVIEW

File: projects/packages/videopress/src/client/admin/components/progress-bar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import type React from 'react';
+
+/**
+ * Progress Bar component
+ *
+ * @param {ProgressBarProps} props - Component props.
+ * @returns {React.ReactNode} - ProgressBar react component.
+ */
+const ProgressBar: React.FC< ProgressBarProps > = ( { className, progress } ) => {
+	if ( progress == null ) {
```
Comment (retrofox) [inline projects/packages/videopress/src/client/admin/components/progress-bar/index.tsx:19]: should it be `===`?

Comment (dhasilva) [reply projects/packages/videopress/src/client/admin/components/progress-bar/index.tsx:19]: `==` tests against `undefined` as well. It is how lodash implements `.isNil`: https://github.com/lodash/lodash/blob/master/isNil.js

File: projects/packages/videopress/src/client/admin/components/video-storage-meter/index.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import type React from 'react';
+
+/**
+ * Video Storage Meter component
+ *
+ * @param {VideoStorageMeterProps} props - Component props.
+ * @returns {React.ReactNode} - VideoStorageMeter react component.
+ */
+const VideoStorageMeter: React.FC< VideoStorageMeterProps > = ( { className, total, used } ) => {
+	if ( ! total || used == null ) {
```
Comment (retrofox) [inline projects/packages/videopress/src/client/admin/components/video-storage-meter/index.tsx:23]:
```suggestion
	if ( ! total || used === null ) {
```

File: projects/packages/videopress/src/client/admin/components/progress-bar/index.tsx (hunk 1 above)
Comment (retrofox) [inline projects/packages/videopress/src/client/admin/components/progress-bar/index.tsx:19]:
```suggestion
	if ( progress === null ) {
```

File: projects/packages/videopress/src/client/admin/components/video-storage-meter/index.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+		return null;
+	}
+
+	const progress = used / total;
+	const progressLabel = `${ ( progress * 100 ).toFixed() }%`;
+	const totalLabel = filesize( total, { base: 2 } );
+
+	return (
+		<div className={ classnames( className ) }>
+			<Text className={ classnames( styles.text ) }>
```
Comment (retrofox) [inline projects/packages/videopress/src/client/admin/components/video-storage-meter/index.tsx]: kind of ambiguous name :-p Please ignore it

File: projects/packages/videopress/src/client/admin/components/video-storage-meter/stories/style.module.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.wrapper {
+  width: 358px;
```
Comment (retrofox) [inline projects/packages/videopress/src/client/admin/components/video-storage-meter/stories/style.module.scss]: I see we are defining width in the stories for testing purposes. I'm not sure about it, considering that it has the screen-size addons:


https://user-images.githubusercontent.com/77539/187513880-06f127c8-cfc6-46fc-8b86-7af927a34375.mov

**Review by retrofox — DISMISSED** [review]
LGTM. I left a few comments, but they are not a big deal. Let's keep in mind this component is a good candidate to get moved to the Jetpack components package. :shipit:

File: projects/packages/videopress/src/client/admin/components/video-storage-meter/index.tsx (hunk 3 above)
Comment (dhasilva) [reply projects/packages/videopress/src/client/admin/components/video-storage-meter/index.tsx]: You mean the `.text`? Yeah. How about something like `.label` or `.percentage`?

File: projects/packages/videopress/src/client/admin/components/video-storage-meter/stories/style.module.scss (hunk 4 above)
Comment (dhasilva) [reply projects/packages/videopress/src/client/admin/components/video-storage-meter/stories/style.module.scss]: Yes. `VideoUploadArea` also does that. I will remove it on both.

File: projects/packages/videopress/src/client/admin/components/video-storage-meter/index.tsx (hunk 3 above)
Comment (retrofox) [reply projects/packages/videopress/src/client/admin/components/video-storage-meter/index.tsx]: `percentage-description`?

File: projects/packages/videopress/src/client/admin/components/video-storage-meter/stories/style.module.scss (hunk 4 above)
Comment (retrofox) [reply projects/packages/videopress/src/client/admin/components/video-storage-meter/stories/style.module.scss]: yeah, I knew you knew it

**Review by retrofox — APPROVED** [review]
LGTM

**Comment by retrofox** [comment]
r251618-wpcom

**Pull request merged**